### PR TITLE
Finish Edit user

### PR DESCRIPTION
**Pull request by littleCitizen6**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ import MyTicketsPage from './Components/Pages/MyTicketsPage/MyTicketsPage.js'
 import CollapsibleTable from './Components/Pages/AllTicketsPhone/AllTickets'
 import CreateGroup from './Components/Pages/Group/CreateGroup.js'
 import Header from './Components/CommonComponents/Navbar/NavBarMaterial'
```
Comment (littlecitizen11) [inline src/App.js:12]: edit the navbar to the page

File: src/App.js [hunk 2]
```diff
@@ -48,21 +49,25 @@ const optionsMapping = {
 }
 
 export default function App() {
-  const [ticketId, setTicketId] = useState('')
+  const [ticketId, setTicketId] = useState('6')
```
Comment (littlecitizen11) [inline src/App.js:52]: what is this mockup?

File: src/App.js [hunk 3]
```diff
@@ -48,21 +49,25 @@ const optionsMapping = {
...
+  const [userActions, setActions] = useState([])
   return (
     <div className="App">
       <Router>
         <Switch>
           <Route path="/" exact component={() => <LoginPage />} />
           <Route path="/Home" exact component={() => <Home optionsMapping={optionsMapping}/>} />
-          <Route path="/EditTicket" component={() => <ViewTicket optionsMapping={optionsMapping}/>} />
+          <Route path="/EditTicket" exact component={() => <ViewTicket ticketId={ticketId} optionsMapping={optionsMapping}/>} />
+          <Route path="/EditUser" exact component={() => <EditUserPage userName="carmel"/>} />
```
Comment (littlecitizen11) [inline src/App.js:62]: mockup?

File: src/Components/Pages/EditUser/EditUser.js [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+            {'.'}
+        </Typography>
+    );
+}
+
+const theme = createMuiTheme({
+    direction: 'rtl',
+});
+
+const useStyles = makeStyles((theme) => ({
```
Comment (littlecitizen11) [inline src/Components/Pages/EditUser/EditUser.js:33]: delete unused props

File: src/Components/Pages/EditUser/EditUser.js [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+        alignItems: 'center',
+        justifyContent:'space-between',
+    }
+}));
+
+
+export default function EditUserPage(props) {
+    const [phoneNumber, setPhoneNumber] = useState();
+    const [password, setPassword] = useState();
+    const [userInfo, setUserInfo] = useState({});
```
Comment (littlecitizen11) [inline src/Components/Pages/EditUser/EditUser.js:63]: you shouldnt use useState to a const var ....
you get the userInfo from server once.. it wount change

File: src/Components/Pages/EditUser/EditUser.js [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+
+
+    const handlePasswordChange = (event) => {
+        // let newPassword = event.target.value;
+        setPassword(event.target.value);
+    };
+
+    const handlePhoneNumberChange = (event) => {
+        //validate phone number
+        let newPhoneNumber = event.target.value;
```
Comment (littlecitizen11) [inline src/Components/Pages/EditUser/EditUser.js:79]: insert the event.target.value into the setPhoneNumber..
you dont need the newPhoneNumber var

File: src/Components/Pages/EditUser/EditUser.js [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+        setPassword("");
+    }
+
+    useEffect(() => {
+        async function Currencies() {
+            const parsed = queryString.parse(window.location.search);
+            let user;
+            try {
+                let getUser = await axios.get(`http://10.1.0.27:3000/Users/ByName/${parsed.userName}`, cookieHeader);
+                user = getUser.data;
```
Comment (littlecitizen11) [inline src/Components/Pages/EditUser/EditUser.js:94]: you dont need those variable.. just set into the useState : user.PhoneNumber and user.data

File: src/Components/Pages/EditUser/EditUser.js [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+    }, [])
+
+    const classes = useStyles();
+
+    const EditUser = async (phoneNumberToEdit, passwordToEdit) => { //hash to password
+        if(changePassword){
+            passwordToEdit = sha256(passwordToEdit)
+            await axios.put(`http://10.1.0.27:3000/User/${userInfo.Username}?phone=${phoneNumberToEdit}&password=${passwordToEdit}`,{}, cookieHeader)
+            .then((res) => {
+                history.push({pathname:'/Home'});
```
Comment (littlecitizen11) [inline src/Components/Pages/EditUser/EditUser.js:110]: you might redirect him agian to the table of the user ?

Comment (littlecitizen11) [reply src/Components/Pages/EditUser/EditUser.js:110]: /Manage

**Review by littlecitizen11 — COMMENTED** [review]
:)

File: src/Components/Pages/EditUser/EditUser.js (hunk 4 above)
Comment (littleCitizen6) [reply src/Components/Pages/EditUser/EditUser.js:33]: done

File: src/Components/Pages/EditUser/EditUser.js (hunk 6 above)
Comment (littleCitizen6) [reply src/Components/Pages/EditUser/EditUser.js:79]: done

File: src/Components/Pages/EditUser/EditUser.js (hunk 8 above)
Comment (littleCitizen6) [reply src/Components/Pages/EditUser/EditUser.js:110]: til

File: src/Components/Pages/EditUser/EditUser.js (hunk 5 above)
Comment (littleCitizen6) [reply src/Components/Pages/EditUser/EditUser.js:63]: yes but i have to give it a value before cause api calls should be in useEffect, so i leave it like this

File: src/Components/Pages/EditUser/EditUser.js (hunk 7 above)
Comment (littleCitizen6) [reply src/Components/Pages/EditUser/EditUser.js:94]: it doesnt work with some of them... with the other done

**Pull request merged**